hash_practice/exercises.py

```diff
@@ -2,19 +2,50 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(N)
+        Space Complexity: O(N)
     """
-    pass
+    if len(strings) == 0:
+        return strings
+
+    anagram_dict = {}
+    for word in strings:
+        sorted_string = "".join(sorted(word))
+        if sorted_string not in anagram_dict:
+            anagram_dict[sorted_string] = [word]
+        else:
+            anagram_dict[sorted_string].append(word)
+    return list(anagram_dict.values())
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
         Time Complexity: ?
         Space Complexity: ?

```

Comment:
Looks like you're missing complexity requirements for this one. I encourage you to think about what it would be for this, but I am still going to mark this Green as your implementation is fine.